api/src/main/java/org/openmrs/module/reporting/report/renderer/XlsReportRenderer.java

```diff
@@ -104,13 +106,26 @@ public void render(ReportData reportData, String argument, OutputStream out) thr
                 for (DataSetColumn column : columnList) {
 					excelBuilder.addCell(column.getLabel(), "bold,border=bottom");
                 }
-                for (DataSetRow row : dataset ) {
-					excelBuilder.nextRow();
-                    for (DataSetColumn column : columnList) {
-                    	Object cellValue = row.getColumnValue(column);
-                        excelBuilder.addCell(cellValue);
-                    }
-                }
+				if (dataset instanceof IterableSqlDataSet) {
```

Comment:
Same comment here as above - I'm curious here to see if we really need to do this check, or if we can use the fact that all datasets have the ability to iterate across the data set rows to mean that we don't need custom logic here, but can leave the nature of the iteration to the internals of the data set implementation.